[MTA] [RN] Release notes for 8.2.0 (#370)
migtools/mta-documentation: docs/topics/release-notes-topics/ref_new-features-and-enhancements-8-2-0.adoc

:_newdoc-version: 2.18.7
:_template-generated: 2026-06-12
:_mod-docs-content-type: REFERENCE

[id="new-features-and-enhancements-8-2-0_{context}"]
= New features and enhancements

[role="_abstract"]
Review new features and enhancements in {ProductFullName} 8.2.0.

> The `mta-ops` command-line interface for stateless workload migrations::

Author: @vashirova, could you please review? TY! :-)

Reviewer: Reviewed, see 2 comments.

> The `mta-ops` command-line interface for stateless workload migrations::

Author: peer reviewed by Valentina, no additional review needed

+

> The `mta-ops` command-line interface (CLI) is a new standalone tool that migrates stateless Red Hat OpenShift workloads. It uses a multi-stage pipeline to locally export, transform, and render your application manifests. As a result, you can safely generate and validate manifests before using standard deployment tools.

Reviewer: I'd consider using {nbsp} between the words "Red" and "Hat", source https://redhat-documentation.github.io/supplementary-style-guide/#non-breaking-spaces.

```suggestion
The `mta-ops` command-line interface (CLI) is a new standalone tool that migrates stateless Red{nbsp}Hat OpenShift workloads. It uses a multi-stage pipeline to locally export, transform, and render your application manifests. As a result, you can safely generate and validate manifests before using standard deployment tools.
```

Or alternatively an attribute for "Red{nbsp}Hat OpenShift".

Author: Correct, great catch! TY! :-)

+

> For more information, see link:[TBD][Migrating stateless workloads by using the mta-ops CLI].

Author: AI: Add the link to the mta-ops doc when ready

> 

Reviewer: Do we need a reference to Jira for this Enh? Ignore if you don't do it for absolutely all RNs.

Author: We don't have a separate Jira for this Enh. We have a few of them instead that cover different features of the new CLI.

> For more information, see link:[TBD][Migrating stateless workloads by using the mta-ops CLI].

Author: ```suggestion
For more information, see link:{mta-URL}/migrating_stateless_workloads_by_using_the_mta_ops_cli/index[Migrating stateless workloads by using the mta-ops CLI].
```

> An OIDC-compliant Hub provider for authentication and authorization::

Reviewer 2: @jortel @sshveta Please could you review the text for OIDC new feature?

The {ProductShortName} Hub implements an OpenID Connect (OIDC) standards-compliant provider that offers local user authentication and federation to other providers. You can use any of the following authentication methods: 

* The Hub OIDC for basic local authentication to reduce infrastructure management and configuration. You do not require any additional configuration to use basic local authentication.
* An LDAP-backed authentication by using a declarative YAML configuration file. 
* An identity provider with {ProductShortName}. 
+
You can integrate Google, Okta, Microsoft Entra ID, or a Keycloak instance you provision independently as an identity provider. If you deployed Keycloak in the {ProductShortName} namespace in earlier versions before an upgrade, {ProductShortName} automatically configures your instance as the identity provider. 
+

> link:TBD[Engineering ticket required]

Reviewer: In the preview, I see the link is attached to the previous paragraph which feels odd. How about we move an empty line before the plus sign to fix indentation?

```suggestion

+
link:TBD[Engineering ticket required]
```

Reviewer 2: You're right...this works for tickets. However, I'm still not very sure if we'll get engineering tickets for new features. So, I've given links to the documentation of these new features. In this case, the links are aligned to the previous paragraph. WDYT, is this ok?

Reviewer: Sure, if that will be a link to docs, I think the current placement is just fine. Thanks

> {ProductShortName} introduces new commands and reorganized subcommands::

Reviewer 2: @eemcmullan  @ibragins 

Please could you review the CLI new feature text?

The {ProductShortName} command-line interface (CLI) introduces a reorganized command model to streamline major tasks you perform by using the CLI. You can use the main command and associated subcommands to complete specific tasks. 
+
link:TBD[Engineering ticket required]

> C# provider compiled with `.Net` Compiler Platform::

Reviewer 2: @mguetta1 Can you please review the C# provider new feature?

> C# provider uses the `.NET` Compiler Platform SDK (Roslyn APIs) for analyses::

Reviewer 3: ```suggestion
C# provider uses the .NET Compiler Platform SDK (Roslyn APIs) for analyses::
```

Reviewer 2: Oh, I'm required to make it bold.

> `C#` provider uses the `.NET` Compiler Platform SDK (Roslyn APIs) for analyses::

Reviewer 2: @dymurray @JonahSussman Please could you review the C# provider enhancement?

> {ProductShortName} supports C# provider compiled with .Net Compiler Platform::

Reviewer 2: @mguetta1 Could you review this C# provider enhancement please?
Also, please share the engineering ticket for this enhancement.

> The `C#` provider uses the `.NET` Compiler Platform software development kit (SDK) with Roslyn application programming interfaces (APIs) to analyze `.NET` and `C#` applications. As a result, you do not need custom rules and external dependencies to analyze `C#` and `.NET` projects. The provider also improves semantic analysis of applications by covering more framework references, object creation, type annotations, and attributes when parsing the source code.

Reviewer 3: ```suggestion
The C# provider uses the .NET Compiler Platform software development kit (SDK) with Roslyn application programming interfaces (APIs) to analyze C# applications. As a result, you do not need external dependencies to analyze C# projects. The provider also improves semantic analysis of applications by covering more framework references, object creation, type annotations, and attributes when parsing the source code.
```

Reviewer 3: I don't think we support any other .NET language analysis besides C#
@dymurray @JonahSussman could you please confirm that?

Reviewer 2: @mguetta1 Please could you confirm why we should remove custom rules in this description? While the user is free to write one, my understanding is that, they do not need to write a custom rule to run an analysis.

Reviewer 3: That's right. The user does not need to write a custom rule to run an analysis, but it has nothing to do with the new roslyn-based implementation.
In fact, the default C# target was already available in 8.1.0, so it cannot be mentioned as an enhancement in 8.2

Reviewer 2: @mguetta1 Can you please approve if the updates look ok to you?

> The `C#` provider uses the `.NET` Compiler Platform software development kit (SDK) with Roslyn application programming interfaces (APIs) to analyze `.NET` applications written in `C#`. As a result, you do not need external dependencies to analyze `C#` projects. The provider also improves semantic analysis of applications by covering more framework references, object creation, type annotations, and attributes when parsing the source code.

Reviewer: I've never seen C# and .NET use backticks. I'd consider removing them.

Reviewer 2: Ah, ok...I'm never too sure about the rules to apply backticks.

:_newdoc-version: 2.18.7
:_template-generated: 2026-06-12
:_mod-docs-content-type: REFERENCE

[id="new-features-and-enhancements-8-2-0_{context}"]
= New features and enhancements

[role="_abstract"]
Review new features and enhancements in {ProductFirstRef} 8.2.0.

> 

Author: ```suggestion
```

:_newdoc-version: 2.18.7
:_template-generated: 2026-06-12
:_mod-docs-content-type: REFERENCE

[id="new-features-and-enhancements-8-2-0_{context}"]
= New features and enhancements

[role="_abstract"]
Review new features and enhancements in {ProductFirstRef} 8.2.0.

// MTA-6885 (multi-stage pipeline)

// MTA-6886 (cluster-level dependencies)

> // MTA-6887 (validation)

Author: @istein1, would a single New feature release note with a link to the full mta-ops doc suffice? Or do you want to have separate release notes for particular features of it? Here's my suggestion: 
```suggestion
The `mta-ops` command-line interface for stateless workload migrations::
+
The `mta-ops` command-line interface (CLI) is a new standalone tool that migrates stateless Red Hat OpenShift workloads. It uses a multi-stage pipeline to locally export, transform, and render your application manifests. As a result, you can safely generate and validate manifests before using standard deployment tools.
+
For more information, see link:[TBD][Migrating stateless workloads by using the mta-ops CLI].
```

Reviewer 4: I think the single-entry approach it better.

Reviewer 4: @mpershina , the suggested content LGTM

Author: Thanks a lot!

> {ProductShortName} offers an OIDC-compliant Hub provider for authentication and authorization::

Reviewer 2: @jortel @sshveta Please could you review the OIDC feature enhancement? Thanks!

To authenticate users in the {ProductShortName} 8.2.0 installations, you can use the Hub OpenID Connect (OIDC) provider. When upgrading to {ProductShortName} 8.2.0, you can opt for any one of the following authentication methods: 

* Use the Hub OIDC for basic local authentication to reduce infrastructure management and configuration. You do not require any additional configuration to use basic local authentication.
* Configure an LDAP-backed authentication by using a declarative YAML configuration file. 
* Configure an identity provider with {ProductShortName}. 
+
You can integrate Google, Okta, Microsoft Entra, or a Keycloak instance you provision independently as an identity provider. If you provisioned Keycloak in the {ProductShortName} namespace in earlier versions before an upgrade, {ProductShortName} will automatically configure your instance as an identity provider. 
+

link:https://redhat.atlassian.net/browse/MTA-6893[MTA-6893]

> {ProductShortName} supports new commands and reorganized subcommands::

Reviewer 2: @eemcmullan
 @ibragins Could you take a look at the enhancement description for CLI commands, please?

:_newdoc-version: 2.18.7
:_template-generated: 2026-06-12
:_mod-docs-content-type: REFERENCE

[id="new-features-and-enhancements-8-2-0_{context}"]
= New features and enhancements

[role="_abstract"]
Review new features and enhancements in {ProductFullName} 8.2.0.

The `mta-ops` command-line interface for stateless workload migrations::
+
The `mta-ops` command-line interface (CLI) is a new standalone tool that migrates stateless Red{nbsp}Hat OpenShift workloads. It uses a multi-stage pipeline to locally export, transform, and render your application manifests. As a result, you can safely generate and validate manifests before using standard deployment tools.
+

> For more information, see link:{mta-URL}/migrating_stateless_workloads_by_using_the_mta_ops_cli/index[Migrating stateless workloads by using the mta-ops CLI].

Author: ```suggestion
For more information, see link:{mta-URL}/migrating_stateless_workloads_by_using_the_mta-ops_cli/index[Migrating stateless workloads by using the mta-ops CLI].
```

* An identity provider with {ProductShortName}. 
+
You can integrate Google, Okta, Microsoft Entra ID, or a Keycloak instance you provision independently as an identity provider. If you deployed Keycloak in the {ProductShortName} namespace in earlier versions before an upgrade, {ProductShortName} automatically configures your instance as the identity provider. 

+
For more information, see link:{mta-URL}/installing_the_migration_toolkit_for_applications/hub-oidc-authentication_installing-mta-title[Authentication and authorization].

+
link:https://redhat.atlassian.net/browse/MTA-7399[MTA-7399]

{ProductShortName} introduces new commands and reorganized subcommands::

The {ProductShortName} command-line interface (CLI) introduces a reorganized command model to streamline major tasks you perform by using the CLI. You can use the main command and associated subcommands to complete specific tasks. 
+ 
For more information, see link:{mta-URL}/using_the_migration_toolkit_for_applications_command-line_interface/about-the-command-line-tool_cli-guide[About the {ProductShortName} commands].

+
link:https://redhat.atlassian.net/browse/MTA-7398[MTA-7398]

> C# provider uses the .NET Compiler Platform (Roslyn) analyzer for analyses::

Reviewer 2: @JonahSussman @dymurray Please review the C# provider enhancement description. Thanks!

Reviewer 5: Looks good!

Reviewer 2: Thank you for your review!
